lib/filesystem/CArchiveLoader.cpp

```diff
@@ -182,4 +212,48 @@ std::unordered_set<ResourceID> CArchiveLoader::getFilteredFiles(std::function<bo
 	return foundID;
 }
 
+void CArchiveLoader::extractToFolder( std::string outputSubFolder, CFileInputStream& fileStream, ArchiveEntry entry)
+{
+	si64 currentPosition = fileStream.tell(); // save filestream position
+
+	std::unique_ptr<char[]> data = std::unique_ptr<char[]>(new char[entry.fullSize]);
```

Comment:
```suggestion
	std::vector<ui8> data(entry.fullSize);
```
And then data.pointer()